lib/plugins/aws/deploy/compile/functions/index.js

```diff
@@ -112,6 +112,17 @@ class AwsCompileFunctions {
         throw new Error(`No artifact path is set for function: ${functionName}`);
       }
 
+      if (this.serverless.service.provider.deploymentBucket) {
```

Comment:
could you please add some validation logic for the bucket name along with some friendly error messages? 😊